sdk/identity/azure-identity/src/token_credential_impl.cpp

```diff
@@ -309,13 +337,14 @@ AccessToken TokenCredentialImpl::ParseToken(
       }
     }
 
+    auto const tzOffsetStr = TimeZoneOffsetAsString(utcDiffSeconds);
     if (expiresOn.is_string())
     {
       auto const expiresOnAsString = expiresOn.get<std::string>();
       for (auto const& parse : {
-               std::function<DateTime(std::string const&)>([](auto const& s) {
+               std::function<DateTime(std::string const&)>([&](auto const& s) {
                  // 'expires_on' as RFC3339 date string (absolute timestamp)
-                 return DateTime::Parse(s, DateTime::DateFormat::Rfc3339);
+                 return DateTime::Parse(s + tzOffsetStr, DateTime::DateFormat::Rfc3339);
```

Comment:
We should check and only add a `tzOffsetStr` **IF** the string doesn't already have a `+XX:XX` info at the end, to avoid malforming the date string.

Reply:
I am assuming this PR will adjust a bit after https://github.com/Azure/azure-sdk-for-cpp/pull/5180 is merged.

Reply:
Azure CLI team is not expected to change format of this field - they will forever keep it as is for compatibility - https://github.com/Azure/azure-cli/issues/19700#issuecomment-1784671351.
If only they could, they would add `Z` and return as UTC timestamp, or add `+HH:MM` of a time zone, this whole sage would've been over, and there would've been no need to add `expires_on` field.

Reply:
Can't `token_credential_impl.cpp` be used for other credentials beyond azure cli credential, in the future?

Reply:
It can and should, however, I don't expect it being used with non-0 `utcDiffSeconds`.
As `ParseToken()`'s comment says, "`No credential other than AzureCliCredential (which gets timestamps as local time without time zone) should need to set it.`"